File: 3. Klasse/Inf/Vehicle/src/Valentin/Adlgasser/Vehicle.java
package Valentin.Adlgasser;

/**
 * This class creates a Vehicle
 */
public class Vehicle {
    private double price;
    private String colour;
    private double weight;
    private int warranty;
    private Motor horsepower;
    private Gearbox gearbox;

    /**
     * Standard constructor
     */
    public Vehicle(double price, String colour, double weight, int warranty, int horsepowers, int gears, boolean automatic) {
        this.price = price;
        this.colour = colour;
        this.weight = weight;
        this.warranty = warranty;
        this.horsepower = new Motor();
        horsepower.setHorsepower(horsepowers);
        this.gearbox = new Gearbox(gears, automatic);
    }

    /**
     * Getters and setters for this class
     */
    public void setHorsepower(int horsepowers) {
        horsepower.setHorsepower(horsepowers);
    }

    public int getHorsepower() {
        return horsepower.getHorsepower();
    }

    public int getWarranty() {
        return warranty;
    }

    public void setWarranty(int warranty) {
        this.warranty = warranty;
    }

    public double getWeight() {
        return weight;
    }

    public void setWeight(double weight) {
        this.weight = weight;
    }

    public String getColour() {
        return colour;
    }

    public void setColour(String colour) {
        this.colour = colour;
    }

    public double getPrice() {
        return price;
    }

    public void setPrice(double price) {
        this.price = price;
    }

    /**
     * @param vat
     * Returns the price + VAT
     */
    public double priceAndVAT(double vat){

        return(this.price + (this.price*(vat/100)));
    }

    /**
     * @param warrantyYears
     * Adds more years to the warranty
     */
    void extendWarranty(int warrantyYears) {

        warranty += warrantyYears;
    }

    /**
     * Outputs the Stats of the vehicle
     */
    void showStats(){
        System.out.println("Price: " + price + "€\r\nColour: " + colour + "\r\nWeight: " + weight + "kg\r\nWarranty: " + warranty + "Years\r\nHorsepower: " + horsepower.getHorsepower() + "\r\n" + gearbox.outputGearbox());
    }
}

File: 2. Klasse/INF/Programme/Vehicle/Vehicle.cpp
//
// Created by valentin.adlgasser on 12.12.2018.
//

#include "Vehicle.hpp"
#include <iostream>

int Vehicle::getWheels() const {
    return wheels;
}

void Vehicle::setWheels(int wheels) {
    Vehicle::wheels = wheels;
}

int Vehicle::getHorsepower() const {
    return horsepower;
}

void Vehicle::setHorsepower(int horsepower) {
    Vehicle::horsepower = horsepower;
}

const int Vehicle::getBuildyear() const {
    return buildyear;
}

void Vehicle::showall() const{
    std::cout << "Wheels: " << wheels << "\n";
    std::cout << "Horsepower: " << horsepower << "\n";
    std::cout << "Buildyear: " << buildyear << "\n";

}

Vehicle::Vehicle(int wheels, int horsepower, const int buildyear) : wheels(wheels), horsepower(horsepower),
                                                                    buildyear(buildyear) {}
